pkg/provider/azure_subnet_repo.go

```diff
@@ -67,3 +67,22 @@ func (az *Cloud) getSubnet(vnetResourceGroup, virtualNetworkName, subnetName str
 	}
 	return subnet, exists, nil
 }
+
+func (az *Cloud) listSubnet(vnetResourceGroup, virtualNetworkName string) ([]network.Subnet, error) {
+	if vnetResourceGroup == "" {
+		if len(az.VnetResourceGroup) > 0 {
+			vnetResourceGroup = az.VnetResourceGroup
+		} else {
+			vnetResourceGroup = az.ResourceGroup
+		}
+	}
+
+	ctx, cancel := getContextWithCancel()
```

Comment:
pls use the context passed by in the caller in `EnsureStorageAccount`, don't create a new context every time.

Reply:
done